app/javascript/hyacinth_v1/components/publish_targets/PublishTargets.js

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import { Route, Switch } from 'react-router-dom';
+import PageNotFound from '../shared/PageNotFound';
+import ProtectedRoute from '../shared/routes/ProtectedRoute';
+import PublishTargetEdit from './PublishTargetEdit';
+import PublishTargetIndex from './PublishTargetIndex';
+import PublishTargetNew from './PublishTargetNew';
+import PublishTargetShow from './PublishTargetShow';
+
+
+function PublishTargets() {
+  return (
+    <Switch>
+      <ProtectedRoute
+        exact
+        path="/publish_targets"
+        requiredAbility={{ action: 'read', subjectType: 'PublishTarget' }}
+        component={PublishTargetIndex}
+      />
+
+      <ProtectedRoute
+        exact
+        path="/publish_targets/new"
+        requiredAbility={{ action: 'create', subjectType: 'PublishTarget' }}
+        component={PublishTargetNew}
+      />
+
+      <ProtectedRoute
+        exact
+        path="/publish_targets/:stringKey"
+        component={PublishTargetShow}
+        requiredAbility={params => ({ action: 'read', subject: 'PublishTarget', publishTarget: { stringKey: params.stringKey } })}
+      />
```

Comment:
curious as to why the publish target stringKey is needed here. can read ability vary based on which public target is being requested?

Reply:
Good question!  With the capabilities of our ability checking system, we technically can assign read permission on a publish target by publish target basis if we wanted to.  Do we actually do that?  No, not at this time.  But I like to be as specific as possible in case we ever change our minds in the future.

A resource like a DigitalObject actually does have id-specific checks, but for a PublishTarget we're more likely to say "user X has read permission on any PublishTarget".  Fortunately, that broader ability automatically covers the more specific stringKey-specific case in checks like the one above.

To be even more accurate though, we're currently only allowing admins to work with publish targets, and for that reason we don't even assign PublishTarget permissions explicitly.  The high level admin permission cascades down like this:

- Admin permission automatically grants broad PublishTarget access for any operation type
- Broad PublishTarget access grants broad PublishTarget read access
- Broad PublishTarget read access automatically grants specific PublishTarget read permission for a given stringKey